example-app/src/main/java/com/adyen/checkout/example/ui/settings/composable/SettingsScreen.kt

```diff
@@ -0,0 +1,267 @@
+            onSwitchSettingChanged = viewModel::onSwitchSettingChanged,
+            modifier = Modifier.padding(innerPadding),
+        )
+    }
+}
+
+@Suppress("LongParameterList")
+@Composable
+private fun SettingsScreen(
+    uiState: SettingsUIState,
+    onItemClicked: (SettingsItem) -> Unit,
+    onEditSettingConsumed: () -> Unit,
+    onTextSettingChanged: (SettingsIdentifier, String) -> Unit,
+    onListSettingChanged: (SettingsIdentifier, EditSettingsData.SingleSelectList.Item) -> Unit,
+    onSwitchSettingChanged: (SettingsIdentifier, Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    SettingsItemsList(
+        settingsCategories = uiState.settingsCategories,
+        onItemClicked = onItemClicked,
+        onSwitchSettingChanged = onSwitchSettingChanged,
+        modifier = modifier,
+    )
+
+    if (uiState.settingToEdit != null) {
+        EditSettingDialog(
+            settingToEdit = uiState.settingToEdit,
+            onTextSettingChanged = {
+                onTextSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onListSettingChanged = {
+                onListSettingChanged(uiState.settingToEdit.identifier, it)
+                onEditSettingConsumed()
+            },
+            onDismiss = {
+                onEditSettingConsumed()
+            },
+        )
+    }
+}
+
+@Composable
+private fun SettingsItemsList(
+    settingsCategories: List<SettingsCategory>,
+    onItemClicked: (SettingsItem) -> Unit,
+    onSwitchSettingChanged: (SettingsIdentifier, Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    LazyColumn(
+        modifier = modifier.fillMaxSize(),
+    ) {
+        settingsCategories.forEachIndexed { index, category ->
+            item {
+                SettingsCategoryHeader(title = stringResource(id = category.titleResId))
+            }
+
+            items(items = category.settingsItems) { settingsItem ->
+                when (settingsItem) {
+                    is SettingsItem.Text -> {
+                        TextSettingsItem(settingsItem, onItemClicked)
+                    }
+
+                    is SettingsItem.Switch -> SwitchSettingsItem(settingsItem, onSwitchSettingChanged)
+                }
+            }
+
+            if (index != settingsCategories.size - 1) {
+                item {
+                    SettingsDivider()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun SettingsCategoryHeader(
+    title: String,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier
+            .padding(
+                start = ExampleTheme.dimensions.grid_2,
+                end = ExampleTheme.dimensions.grid_2,
+                top = ExampleTheme.dimensions.grid_2,
+            )
+            .fillMaxWidth(),
+    ) {
+        Text(
+            title,
+            style = MaterialTheme.typography.labelLarge,
```

Comment:
Can we use a larger type here? To make the header and items more distinguishable